builtin/logical/database/path_roles.go

```diff
@@ -210,6 +210,12 @@ func (b *databaseBackend) pathRoleCreateUpdate() framework.OperationFunc {
 			} else if req.Operation == logical.CreateOperation {
 				role.Statements.Renewal = data.Get("renew_statements").([]string)
 			}
+
+			// Do not persist deprecated statements that are populated on role read
```

Comment:
What if we are only updating TTLs,  in that case i don't think we would want to always wipe these values, right? Should we only update them if we overwrite them above?

Reply:
Could you also add a test for this case ☝️

Reply:
I will definitely add a test for it, but as is I _think_ we're good. The b.Role() statement at the top of this function is always going to return a role populated with both the old and new versions, even if the old version was all that existing previously. We shouldn't have any reason to save the old if we have the new, and not saving the old breaks the upgrade cycle that was causing the original problem.

Reply:
Tests were added for TTL-only updates and things are working as expected.